triton/shard_record.go

```diff
@@ -0,0 +1,11 @@
+package triton
+
+type shardRecord struct {
+	record   map[string]interface{}
+	shard    string
+	sequence string
```

Comment:
I've got types specifically for these to make it less confusing. I think this would fit in better as:

    type shardRecord struct {
        ....
        shardID ShardID
        sequenceNumber SequenceNumber
    }

Just like in https://github.com/postmates/go-triton/blob/master/triton/stream.go#L21